src/lock/postgres.rs

```diff
@@ -0,0 +1,182 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+use anyhow::{Context, Result};
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+use tracing::{debug, warn};
+
+use super::DistributedLock;
+
+/// PostgreSQL-based distributed lock using advisory locks.
+///
+/// This lock uses PostgreSQL advisory locks to coordinate access across multiple hosts.
+/// The lock is identified by a 64-bit integer derived from hashing the target path.
+pub struct PostgresLock {
+    client: Arc<Mutex<Client>>,
+    lock_key: i64,
+    target_path: PathBuf,
+    owner: Option<String>,
+    acquired_at: SystemTime,
+}
+
+impl PostgresLock {
+    /// Create a new PostgreSQL lock client.
+    ///
+    /// # Arguments
+    /// * `connection_string` - PostgreSQL connection string (e.g., "host=localhost user=postgres")
+    pub async fn connect(connection_string: &str) -> Result<Arc<Mutex<Client>>> {
+        let (client, connection) = tokio_postgres::connect(connection_string, NoTls)
```

Comment:
I'm ok we use tokio_postgres as it's widely used. but if possible could we use duckdb's postgres extension to connect, then we don't need extra dependencies. we can refer to engine/factory.rs:

```
let conn = Connection::open_in_memory_with_flags(config)?;
conn.execute_batch("ATTACH '' AS lock_db (TYPE postgres); use lock_db;")?;
Ok(conn)
```

this way we will get a conn, and we can use `select` or commands like ordinary.